package.json

```diff
@@ -1,16 +1,14 @@
 {
   "private": true,
-  "scripts": {
-    "test": "spago test",
-    "build": "spago build",
-    "bundle": "spago bundle-app --to dist/app.js",
-    "bundle:watch": "webpack --mode=development --entry ./entry.js --output-path ./dist --output-filename app.js --progress --watch"
-  },
   "devDependencies": {
+    "parcel": "^1.12.4",
     "purescript": "^0.13.8",
-    "purescript-psa": "^0.7.3",
-    "spago": "^0.15.2",
-    "webpack": "^4.43.0",
-    "webpack-cli": "^3.3.11"
+    "spago": "^0.15.3"
+  },
+  "scripts": {
+    "build": "spago build",
+    "serve": "parcel dev/index.html --out-dir dev-dist --open",
+    "prep-prod": "mkdir -p prod && cp dev/index.html prod/ && rm -rf dist",
+    "build-prod": "npm run prep-prod && spago bundle-app --to prod/index.js && parcel build prod/index.html"
```

Comment:
I'd like to avoid multi-stage commands if we can. Are we able to get away with `build-prod` alone?

Reply:
Also, is it possible to just have a single `dist` directory rather than several possible output directories?

Reply:
> Are we able to get away with build-prod alone?

Sure. We can move `prep-prod` contents into `build-prod`. I just broke that up to clarify intent. I don't believe npm scripts allow line wrapping though, so this either needs to be one big line, or moved to a shell script. Any preference?

> is it possible to just have a single dist directory

`dev/index.js` is setup for fast hot-reloading, while `prod/index.js` is generated to produce the smallest bundles. I don't know of a simpler way to organize this project to achieve both of those goals. Open to suggestions.

Reply:
Oh, I may have misread your last suggestion. The reason for a separate `dev-dist` is so users don't accidentally try to upload the unminified output.

Reply:
> so this either needs to be one big line, or moved to a shell script. Any preference?

If possible I'd like to keep the commands concise and in the package.json, without shell scripts. I think we view this template as a starting point for building a Halogen project and expect that you will scaffold however you want to organize your project on top of it, whether you want to use shell scripts or a Makefile or a Nix expression or whatever else. That's in contrast to something like `create-react-app` which makes a lot of opinionated decisions for you.

With that in mind I'd like to keep the commands in the package.json even if the line lengths get a little long.

> The reason for a separate dev-dist is so users don't accidentally try to upload the unminified output. 

Can we have a single `dist` where the unminified output is gitignored and the minified output is not?

Reply:
> I'd like to keep the commands in the package.json even if the line lengths get a little long.

I'll squish these together then.

> where the unminified output is gitignored and the minified output is not

The .gitignore strategy wouldn't help with deployment strategies where you just select a directory to upload (not necessarily via git).

One workaround is to always clean the dist dir before prod builds, but I recall some other annoying corner cases with this strategy (perhaps if the dev server is kept alive). I'll dig into this some more.